### DISCUSSION TITLE
Issue fitting LGCP Model

### DISCUSSION THREAD

**Discussion by lvclarke** [inlabru-org/inlabru#197] [Q&A] [answered]
Hi, 

I'm running into an issue when fitting my LGCP to distance sampling data, code used below. Suspect it may be that model is not converging? 

# Specify truncation distance of 80m
W <- 80

bru_options_set(bru_verbose = 3)
bru_options_set(verbose = TRUE)

# Remove sightings outside the specified maximum distance
message(paste0("Sightings outside W=", W, ": ", sum(nests$distance >= W)))
message(paste0("Sightings inside W=", W, ": ", sum(nests$distance < W)))
nests_ <- nests[nests$distance < W, , drop = FALSE]

# Specify half-normal detection function
hn <- function(distance, lsig) {exp(-0.5 * (distance / exp(lsig))^2)}

# Define a (Matern) covariance function for the SPDE
matern <- inla.spde2.pcmatern(mesh10,
                              prior.sigma = c(2, 0.01),
                              prior.range = c(50, 0.01)
)

# Specify components of model
cmp <- ~ mySPDE(main = coordinates, model = matern) + lsig(1) + Intercept(1)

# Specify formula
form <- coordinates + distance ~ mySPDE +
  log(hn(distance, lsig)) +
  Intercept + log(2)

# Fit model
fit <- lgcp(
  components = cmp,
  data = nests_,
  samplers = transects,
  domain = list(
    coordinates = mesh10,
    distance = INLA::inla.mesh.1d(seq(0, W, length.out = 20))
  ),
  formula = form,
  options = list(bru_initial = list(lsig = 2),
                 bru_verbose=4)
)

The error I get is:
iinla: Iteration 1 [max:10]
/Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/INLA/bin/mac/64bit/inla.run: line 134: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/INLA/bin/mac/64bit/inla: No such file or directory
/Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/INLA/bin/mac/64bit/inla.run: line 140: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/INLA/bin/mac/64bit/inla: No such file or directory
/Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/INLA/bin/mac/64bit/inla.run: line 140: exec: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/INLA/bin/mac/64bit/inla: cannot execute: No such file or directory
Error in inla.inlaprogram.has.crashed() : 
  The inla-program exited with an error. Unless you interupted it yourself, please rerun with verbose=TRUE and check the output carefully.
  If this does not help, please contact the developers at <help@r-inla.org>.

 *** inla.core.safe:  inla.program has crashed: rerun to get better initial values. try=1/2 
/Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/INLA/bin/mac/64bit/inla.run: line 134: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/INLA/bin/mac/64bit/inla: No such file or directory
/Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/INLA/bin/mac/64bit/inla.run: line 140: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/INLA/bin/mac/64bit/inla: No such file or directory
/Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/INLA/bin/mac/64bit/inla.run: line 140: exec: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/INLA/bin/mac/64bit/inla: cannot execute: No such file or directory
Error in inla.inlaprogram.has.crashed() : 
  The inla-program exited with an error. Unless you interupted it yourself, please rerun with verbose=TRUE and check the output carefully.
  If this does not help, please contact the developers at <help@r-inla.org>.

 *** inla.core.safe:  inla.program has crashed: rerun to get better initial values. try=2/2 
/Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/INLA/bin/mac/64bit/inla.run: line 134: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/INLA/bin/mac/64bit/inla: No such file or directory
/Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/INLA/bin/mac/64bit/inla.run: line 140: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/INLA/bin/mac/64bit/inla: No such file or directory
/Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/INLA/bin/mac/64bit/inla.run: line 140: exec: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/INLA/bin/mac/64bit/inla: cannot execute: No such file or directory
Error in inla.inlaprogram.has.crashed() : 
  The inla-program exited with an error. Unless you interupted it yourself, please rerun with verbose=TRUE and check the output carefully.
  If this does not help, please contact the developers at <help@r-inla.org>.
Warning in iinla(model = info[["model"]], lhoods = info[["lhoods"]], options = info[["options"]]) :
  iinla: Problem in inla: Error in inla.core.safe(formula = formula, family = family, contrasts = contrasts,  : 
  *** Fail to get good enough initial values. Maybe it is due to something else.
iinla: Problem in inla: 1: lgcp(components = cmp, data = nests_, samplers = transects, domain = list( [...]
       distance = INLA::inla.mesh.1d(seq(0, W, length.out = 20))), 
       formula = form, options = list(bru_initial = list(lsig = 2), 
           bru_verbose = 4))
2: bru(components, lik, options = options)
3: iinla(model = info[["model"]], lhoods = info[["lhoods"]], options = info[[ [...]
4: try_callstack(...)
5: do.call(INLA::inla, inla.options.merged, envir = environment(model$effects))
6: (function (formula = NULL, family = "gaussian", contrasts = NULL, 
       data = NULL, quantiles = c(0.025, 0.5, 0.975), E = NULL, 
       offset = NULL, scale = NULL, weights = NULL, Ntrials = NULL, 
       strata = NULL, lp.scale = NULL, link.covariates = NULL, verbose = inla [...]
       lincomb =  [... truncated]
iinla: Giving up and returning last successfully obtained result for diagnostic purposes.

Seems like it may be crashing? Any idea what may be going on here?

Any help greatly appreciated thank you!

**Comment by finnlindgren** [comment, marked as answer]
The initial error message
```
/Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/INLA/bin/mac/64bit/inla.run: line 134: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/INLA/bin/mac/64bit/inla: No such file or directory
```
means that the INLA package isn’t properly installed.
Try reinstalling it, restart R, and then run the code again.

**Comment by finnlindgren** [comment]
Also let me know what inla version it is.

**Reply by lvclarke** [reply]
Hi Finn, thanks so much for such a quick reply. This worked! I'm using inla 23.06.29 and inlabru 2.8.0. I had been fiddling around with installing the development versions before this so assume I must have done something wrong with that, but a reinstall and R restart sorted it.
